src/ascent/runtimes/flow_filters/ascent_runtime_param_check.cpp

```diff
@@ -232,6 +275,85 @@ path_helper(std::vector<std::string> &paths,
 
 }
 
+template<typename T>
+T conduit_cast(const conduit::Node &node);
+
+template<>
+int conduit_cast<int>(const conduit::Node &node)
+{
+  return node.to_int32();
+}
+
+template<>
+double conduit_cast<double>(const conduit::Node &node)
+{
+  return node.to_float64();
+}
+
+template<>
+float conduit_cast<float>(const conduit::Node &node)
+{
+  return node.to_float32();
+}
+
+template<typename T>
+T get_value(const conduit::Node &node, DataObject *dataset)
+{
+  T value = 0;
+  // check to see if this is an expression
+  if(node.dtype().is_string())
+  {
+    if(dataset == nullptr)
+    {
+      ASCENT_ERROR("Numeric parameter is an expression(string)"
+                   <<" but we can not evaluate the expression."
+                   <<" This is usaully for a parameter that is "
+                   <<"not meant to have an expression. expression '"
+                   <<node.to_string()<<"'");
```

Comment:
i guess we always need the dataset ptr, even if the expression doesn't invoke anything requiring it? 

Like: `1+1`

sure -- that seems like a goofy use case, but I could see folks wanting to use trig funcs, etc?

Reply:
if we want to do this, future PR would be best, just wanted to comment while thinking about it.

Reply:
That is why I suggested creating a way to invoke the expression eval without a data set. That said, if we don't give it a data set, the there are a class of functions that will not be available. I suppose we can always give it access to the source.

Reply:
One suboptimal thing is that if we evaluated the expression inside a pipeline, converting from vtkm -> conduit is probably making a copy, even if the expression was `1+1`.

Reply:
I take that back. Looks like we do honor the zero copy in this case.